Fix Tauri build

BrunoBernardino opened the pull request:
This isn't working, but has some attempts. It's currently blocked, because the built app just has many `ReferenceError: Can't find variable: process`. Something with Webpack isn't building properly, as even trying to just run `webpack serve` from inside yields the same result (I updated webpack, but the problem is in the setup/config, really). I also tried setting `resolve.fallback.process: require.resolve('process/browser')` ([among other things](https://pretagteam.com/question/webpack-5-uncaught-referenceerror-process-is-not-defined)), but the executed code comes from `@padloc/app`, so I'd have to add that package there, and it feels wrong. I'm positive I'm just doing something wrong with the setup, but already spent quite some time on it.

To test, pull this down, run `npm install`, and `npm run tauri:dev` (runs with `webpack` instead of `tauri`, just because the console errors are more useful).

In the process of trying to get this to work, I migrated the icons to the assets directory and updated them. I also tweaked the initial HTML to match the new v4 theme.

Related to #281 

Since this is currently blocked, unless we change this to just load the PWA (like electron), thoughts are welcome @MaKleSoft !

BrunoBernardino commented on packages/tauri/webpack.config.js:
```diff
@@ -48,24 +49,29 @@ module.exports = {
     },
     plugins: [
         new EnvironmentPlugin({
+            PL_APP_NAME: name,
```
These missing variables were breaking the app. I kept the defaults the same as PWA.

BrunoBernardino commented:
I tested the app locally and found no issues (with either local storage or clipboard), except that the app's text looks blurry (I have a scaled UI, and don't experience this on electron, for example). I didn't implement the updater yet because it'll require a new server endpoint, and since I haven't looked much into the server, I figured you might have an opinion on how to build that a new endpoint (as explained in https://tauri.studio/en/docs/usage/guides/updater).

I think it's ready for a first review!

BrunoBernardino commented:
Judging by https://github.com/padloc/padloc/runs/4642001378?check_suite_focus=true it seems something's wrong with the `icns` file I built in Linux. I might need to jump into a mac to do it, if you can't.

MaKleSoft commented:
Hm, now both Windows and Linux builds fail saying they can't find any icon at all. It does work on macos though, stragely. @BrunoBernardino Can you run the build locally and check if the `tauri-src/icons` directory is created correctly?

BrunoBernardino commented:
Fun! It does create all the icons locally in Linux, and they "appear fine", except for the ".ico" (which my system supports and shows for other .ico files):
![Screenshot from 2021-12-28 10-57-55](https://user-images.githubusercontent.com/1239616/147559432-a32e96ad-eb87-4
![Screenshot from 2021-12-28 10-58-56](https://user-images.githubusercontent.com/1239616/147559496-b268ff6d-b453-40f3-a6ae-3392f8e0a9dd.png)
cd9-82e8-4ef62443f405.png)

I do get the exact same error that happens in CI:

```
thread 'main' panicked at 'couldn't find a square icon to use as AppImage icon', /home/runner/work/binary-releases/binary-releases/tooling/bundler/src/bundle/linux/appimage.rs:61:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
lerna ERR! npm run build exited 1 in '@padloc/tauri'
```

Given that, I'll keep this command as expected for macos, and just copy the files from assets for linux and windows.

MaKleSoft commented:
Maybe it's a race condition or something? Does it build when you try it a second time with the `icons` dir already generated?

MaKleSoft commented:
Actually I just noticed that if I run it from the root directory using `npm run tauri:build`, the build succeeds by the icon is "empty".
<img width="313" alt="Screenshot 2021-12-28 at 12 25 10" src="https://user-images.githubusercontent.com/1270940/147561542-68d09d5a-12ba-4f5a-8946-8a1dd789290b.png">
Let me try changing the npm script to cd into the package directory instead of executing via lerna

BrunoBernardino commented:
Good catch! I just pushed a fix where I got it to build locally for Linux. I also added a `npm run setup-tauri` to be run inside `packages/tauri` for when someone updates the `Cargo.lock`, to have their system build with those (I imagine you'll see some changes if you run that now, id you had tauri setup from before this branch's changes)

MaKleSoft commented:
The mac app loads successfully now, but I'm having trouble logging in. It does work when I build locally though. It would be nice to be able to bring up the inspector in nightly/beta builds. Can we use the `--debug` flag in all environments except production?

BrunoBernardino commented:
@MaKleSoft I hadn't tried to login, but ran into some CORS issues. I pushed that fix and the beta artifacts should be available in ~10 minutes at https://github.com/padloc/padloc/actions/runs/1631644671

I also made the non-prod builds include the console (`--debug`), like you mentioned.

MaKleSoft reviewed (changes requested):
Looking good! Tested on both MacOS and Windows and don't see any major problems. Keyboard shortcuts (quit, copy, paste etc.) don't work on MacOS but that should get resolved with https://github.com/padloc/padloc/issues/364.

Once you've addressed my pending comments I think we're good to merge.

MaKleSoft commented:
P.S.: Can you try to add a signed & notarised build for macos as well? Should be pretty straightforward: https://tauri.studio/en/docs/usage/guides/bundler/sign-osx. If you into problems we can defer it to another PR.

BrunoBernardino commented on packages/tauri/package.json:
```diff
@@ -6,7 +6,8 @@
         "build": "tauri build",
-        "dev": "tauri dev"
+        "dev": "tauri dev",
+        "update": "cd src-tauri && cargo update -p tauri"
```
Good catch, they're both the same. I'll change for consistency.

BrunoBernardino commented on packages/tauri/package.json:
```diff
@@ -5,8 +5,11 @@
+        "build:production": "tauri build",
+        "dev": "tauri dev",
+        "update": "cd src-tauri && cargo update -p tauri",
+        "setup-tauri": "cd src-tauri && cargo build"
```
The problem is that if someone updates `Cargo.lock` (by running the update deps command, for example), nothing updates those packages (like `npm install`) for others with the same code pulled down.

If you're not freshly installing `tauri` (which we don't install when people run `npm install` from the root, as it's an "external dependency", like "cordova", which also requires `rust`), you would see the new lockfile, but keep using the old dependencies.

This command makes it easy for someone to update their local dependencies to the `Cargo.lock`. If one just ran the update command, they'd potentially/likely override the dependencies too (and the `Cargo.lock` file).

I hope that makes sense.

MaKleSoft replied:
Got it. Thanks for the explanation!

BrunoBernardino commented on packages/tauri/src/index.html:
```diff
@@ -1,21 +1,25 @@
         />
+        <meta name="apple-mobile-web-app-capable" content="yes" />
+        <link rel="shortcut icon" type="image/png" href="/favicon.png" />
+        <link rel="apple-touch-icon" href="/favicon.png" />
```
Not really. I'll remove it.

BrunoBernardino commented:
@MaKleSoft I addressed all your points and confirmed the beta build for linux still works (I have copy/pasting working as well, still had it).

> P.S.: Can you try to add a signed & notarised build for macos as well?

I mentioned this before, because that'll require another action. I'll remind you I also did not implement the updater because of the server requirements. Since you didn't say anything about it, I'll just quote my comment, so we can either create an issue about it, or make a decision:

> I didn't implement the updater yet because it'll require a new server endpoint, and since I haven't looked much into the server, I figured you might have an opinion on how to build that a new endpoint (as explained in https://tauri.studio/en/docs/usage/guides/updater). I didn't implement the signing because that requires the tauri publishing step, which we'll add in another PR/action, for the releases.

Beta artifacts will be live at https://github.com/padloc/padloc/actions/runs/1633700651

MaKleSoft reviewed (approved):
Ok, let's merge this. We can take care of the code signing when we implement the release flow. Auto updates are tracked here: https://github.com/padloc/padloc/issues/365
